pkg/controller/statefulset/stateful_set_control.go

```diff
@@ -667,22 +674,26 @@ func (ssc *defaultStatefulSetControl) rollingUpdateStatefulsetPods(
 		}
 
 		// delete the Pod if it is not already terminating and does not match the update revision.
-		if !isTerminating(replicas[target]) {
+		if !specifiedDeletedPods.Has(replicas[target].Name) && !isTerminating(replicas[target]) {
 			// todo validate in-place for pub
 			inplacing, inplaceUpdateErr := ssc.inPlaceUpdatePod(set, replicas[target], updateRevision, revisions)
 			if inplaceUpdateErr != nil {
 				return status, inplaceUpdateErr
 			}
+			// if pod is inplacing or actual deleting, decrease revision
+			revisionNeedDecrease := inplacing
```

Comment:
if using  inplace-update, the pod can still be stuck in the pre-update hook

Reply:
How about creating a new pull request (PR) to address the pre-updating issue? 
We should also take a look at this CloneSet and DaemonSet.

Reply:
we create a new issue about it: https://github.com/openkruise/kruise/issues/1746